frontend/src/frontend/code_generator/user_callgraph.py

```diff
@@ -30,6 +33,9 @@ def __init__(self, functions: Dict[int, blocks.Function], entry_point: int,
         for function in self.functions.values():
             for cb in function.code_blocks:
                 self.code_blocks[cb.name] = cb
+        if cacheline is None:
+            cacheline = blocks.CacheLine()
+        self.cacheline = cacheline
```

Comment:
Since Cacheline is a global property, I don't think it needs to be stored as an instance variable. It might make better sense as a global variable.